core/shark_turbine/kernel/_support/indexing.py

```diff
@@ -122,20 +115,31 @@ class Grid(metaclass=_GridMeta, symbolic_shape=None):
     """Grid with bounding symbolic shape information in the type."""
 
     symbolic_shape: ClassVar[Optional[SymbolicShapeExpr]]
+    # TODO: dims should also allow dynamic dimensions.
+    dims: list[int]
     rank: int
 
-    def __init__(self, *dims: int):
-        rank = len(dims)
-        if self.symbolic_shape is not None:
-            if rank != len(self.symbolic_shape):
-                raise ValueError(
-                    f"Cannot create {type(self)}({', '.join(str(i) for i in dims)}): mismatched symbolic rank"
-                )
+    def __init__(self):
+        # Resolve the symbolic shape to concrete values.
+        idxc = IndexingContext.current()
+        if self.symbolic_shape:
+            dims = [idxc.get_static_value(dim) for dim in self.symbolic_shape]
+            if None in dims:
+                raise ValueError(f"NYI: Dynamic dims in Grid")
+            self.dims = cast(list[int], dims)
+        else:
+            self.dims = []
 
-        self.dims = dims
         # Shadow the type rank with the actual, which makes it concrete
         # for the generic case.
-        self.rank = rank
+        self.rank = len(self.dims)
+
+    def __class_getitem__(
```

Comment:
This is just to fix a type error. Metaclasses should not have __class_getitem__ on them.